File: Homeworks/C#2/StringsAndTextProcessingHomework/03.CorrectBrackets/CorrectBrackets.cs
/*
Problem 3. Correct brackets

Write a program to check if in a given expression the brackets are put correctly.
Example of correct expression: ((a+b)/5-d). Example of incorrect expression: )(a+b)).
*/ 


using System;

namespace CorrectBracketsProblem
{
    class CorrectBrackets
    {
        static bool AreBracketsCorrect(string expression)
        {
            int bracketsCounter = 0;
            foreach (var element in expression)
            {
                if (element == '(')
                {
                    bracketsCounter++;
                }
                else if (element == ')')
                {
                    bracketsCounter--;
                }

                if (bracketsCounter < 0)
                {
                    return false;
                }
            }

            return bracketsCounter == 0;
        }

        static void Main()
        {
            Console.Write("Please enter an expression: ");
            string input = Console.ReadLine();

            if (AreBracketsCorrect(input) == true)
            {
                Console.WriteLine("The brackets in expression are put correctly");
            }
            else
            {
                Console.WriteLine("The brackets in your expression are put incorrectly");
            }
        }
    }
}

File: Homeworks/JavaScriptFundamentals/08.Strings/02.CorrectBrackets/CorrectBrackets.js
/*
 Write a JavaScript function to check if in a given expression the brackets are put correctly.
 Example of correct expression: ((a+b)/5-d). Example of incorrect expression: )(a+b)).
 */

var jsConsole,
    firstExpression = '((a+b)/5-d)',
    secondExpression = ')(a+b))',
    brackets = 0;


function isBracketsCorrect(expression) {
    var isCorrect = false;

    for (var i = 0; i < expression.length; i += 1) {

        if (expression[i] === '(') {
            brackets += 1;
        }
        else if (expression[i] === ')') {
            brackets -= 1;
        }
    }

    if (brackets === 0) {
        isCorrect = true;
    }

    return isCorrect;
}

jsConsole.writeLine('Is brackets correct: ' + isBracketsCorrect(firstExpression));
console.log('Is brackets correct: ' + isBracketsCorrect(firstExpression));

jsConsole.writeLine('Is brackets correct: ' + isBracketsCorrect(secondExpression));
console.log('Is brackets correct: ' + isBracketsCorrect(secondExpression));